c++/src/kj/memory.h

```diff
@@ -656,6 +657,155 @@ class SpaceFor {
   };
 };
 
+#define KJ_ASSERT_PIN_PTRS
```

Comment:
rather than have an assertion I can make:

```
template <typename T, bool ChecksEnabled = true>
class Pin
```
 to be able to use both.

Reply:
I like the template better than the macro since I can have some `Pin`s checked but not all.

On the other hand, I don't think I'd ever want a `Pin<T, false>`. If I want a bare pointer, I can just  make a `T` and take its address. If I'm going to the trouble of constructing a `Pin<T>`, then it's because I want safety.

Reply:
Given the choice of these two options, I'd agree that the define (and the ability to turn off the checks entirely in a release build) is likely the better choice. That said, we could do both.

```
template <typename T, bool AlwaysChecked = false>
class Pin;
```

When `AlwaysChecked` is `false`, we fallback to the define. If `true`, we always check this instance regardless of define.

It adds a bit more complexity so not sure if it's a more desirable approach, but it's an option. May be useful as an escape hatch for cases where we really want to check for extra safety. 